test/helpers/test-runner/json-test-runner.ts

```diff
@@ -4,10 +4,16 @@ import { expect } from '../../setup'
 import { ethers } from 'hardhat'
 import { Contract, BigNumber } from 'ethers'
 
-const bigNumberify = (arr) => {
+const bnRegex = /^\d+n$/gm
```

Comment:
Made this generally more reliable. Treat more things as big numbers so that equality checking is easier.